master/master.cfg

```diff
@@ -570,33 +570,19 @@ def create_win_factory(os, llvm):
       target_env = env.copy()
       target_env['HL_TARGET'] = hl_target
       target_env['HL_JIT_TARGET'] = hl_target
-      factory.addStep(
-        ShellCommand(name = 'correctness',
-                     description = 'correctness',
-                     locks = [performance_lock.access('counting')],
-                     workdir = 'halide-build-Release',
-                     env = target_env,
-                     timeout = 3600,
-                     command = ['cmd', '/c',
-                                'for', '%i', 'in', '(bin\\Release\\correctness*.exe)', 'do', '(%i)']))
-      factory.addStep(
-        ShellCommand(name = 'performance',
-                     description = 'performance',
-                     locks = [performance_lock.access('exclusive')],
-                     workdir = 'halide-build-Release',
-                     env = target_env,
-                     timeout = 3600,
-                     command = ['cmd', '/c',
-                                'for', '%i', 'in', '(bin\\Release\\performance*.exe)', 'do', '(%i)']))
-      factory.addStep(
-        ShellCommand(name = 'generators',
-                     description = 'generators',
-                     locks = [performance_lock.access('counting')],
-                     workdir = 'halide-build-Release',
-                     env = target_env,
-                     timeout = 3600,
-                     command = ['cmd', '/c',
-                                'for', '%i', 'in', '(bin\\Release\\generator*.exe)', 'do', '(%i)']))
+      for testgroup in ['correctness', 'performance', 'generators', 'tutorials']:
+        factory.addStep(
+          ShellCommand(name = testgroup,
+                       description = testgroup,
+                       locks = [performance_lock.access('counting')],
```

Comment:
This looks like it is changing the locks from 'counting' to 'exclusive' for the performance tests. Is that intended?

Reply:
Oh, good catch. The performance tests need to hold the lock in exclusive mode.

Reply:
Good catch, I was sloppy in my generalization.

Reply:
PTAL